frontend/src/app/components/label-explorer/label-explorer.component.ts

```diff
@@ -46,33 +57,21 @@ export class LabelExplorerComponent implements OnInit {
         this.emitLabelChange(label);
     }
 
-    public removeLabel(sliceId: number, tagKey: string, tool: string): void {
-        const tag: LabelTag = this.tags.find((item: LabelTag) => item.key === tagKey && item.tools.includes(tool));
-
-        if (tag) {
-            const index = this.labels.findIndex(
-                (item: LabelListItem) =>
-                    (item.sliceIndex === sliceId) &&
-                    (item.tag === tag)
-            );
-            if (index > -1) {
-                this.labels.splice(index, 1);
-            } else {
-                console.warn(`LabelExplorerComponent | removeLabel: cannot find label for sliceId: ${sliceId} and tag: ${tag}`);
-            }
+    public removeLabel(selectionId: number): void {
+        const index = this.labels.findIndex((item) => item.selectionId === selectionId);
+        if (index !== -1) {
+            this.labels.splice(index, 1);
         } else {
-            console.warn(`LabelExplorerComponent | removeLabel: cannot find tag for key: ${tagKey} and tool: ${tool}`);
+            console.warn(`LabelExplorerComponent | removeLabel: cannot find label for selectionId ${selectionId}`);
         }
     }
 
     // TODO: tagKey should be part of dict stored in backend (labelling context)
     // TODO: tools should be part of dict stored in backend (available tools)
-    public addLabel(labelSlice: number, tagKey: string, tool: string): void {
+    public addLabel(selectionId: number, labelSlice: number, tagKey: string, tool: string): void {
         const tag: LabelTag = this.getLabelTag(tagKey, tool);
-        const newItem: LabelListItem = new LabelListItem(labelSlice, tag);
-        if (!this.labels.find(label => label.sliceIndex === newItem.sliceIndex)) {
-            this.labels.push(new LabelListItem(labelSlice, tag));
-        }
+        const newItem: LabelListItem = new LabelListItem(selectionId, labelSlice, tag, tool);
+        this.labels.push(newItem);
```

Comment:
Now we have exacly the same labels for different selections on slice. We need to render some numbers or other ids to tell apart labels from each other. 
![opera_2018-07-01_14-15-26](https://user-images.githubusercontent.com/15706349/42134301-3f53d47a-7d39-11e8-8394-d5d75e17130e.png)